src/AggregateFunctions/UniqExactSet.h

```diff
@@ -143,13 +134,12 @@ class UniqExactSet
                     };
 
                     for (size_t i = 0; i < std::min<size_t>(thread_pool->getMaxThreads(), rhs.NUM_BUCKETS); ++i)
-                        thread_pool->scheduleOrThrowOnError(thread_func);
-                    thread_pool->wait();
+                        runner(thread_func, Priority{});
+                    runner.waitForAllToFinishAndRethrowFirstError();
                 }
                 catch (...)
                 {
-                    thread_pool->wait();
-                    throw;
+                    runner.waitForAllToFinishAndRethrowFirstError();
```

Comment:
Does it make sense to set is_cancel before we wait here?

Reply:
I think so